viewing_party/party.py

```diff
@@ -0,0 +1,154 @@
+from collections import Counter
+
+
+def create_movie(title, genre, rating):
+    if title is None:
+        return None
+    if genre is None:
+        return None
+    if rating is None:
+        return None
+
+    movie = {}
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+
+    return movie
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            add_to_watched(user_data, movie)
+    return user_data
+
+
+def get_watched_avg_rating(user_data):
+    ratings_list = []
+
+    if len(user_data["watched"]) == 0:
+        return 0.0
+
+    for movie in user_data["watched"]:
+        ratings_list.append(movie["rating"])
+
+    average_rating = sum(ratings_list) / len(ratings_list)
+    return average_rating
+
+
+def get_most_watched_genre(user_data):
+    genre_list = []
+
+    if user_data["watched"] == []:
+        return None
+
+    for movie in user_data["watched"]:
+        genre_list.append(movie["genre"])
+
+    occurence_count = Counter(genre_list)
+    return occurence_count.most_common(1)[0][0]
+
+
+def get_unique_watched(user_data):
+    user_set = set()
+    friends_set = set()
+    user_list = []
+
+    for movie in user_data["watched"]:
+        user_set.add(movie["title"])
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_set.add(movie["title"])
+
+    user_watched_movies_not_by_friends = user_set - friends_set
+
+    for movie in user_watched_movies_not_by_friends:
+        user_list.append({"title": movie})
```

Comment:
Even though this wave only has title in the movie dictionary, notice that previous and subsequent waves had other fields as well. If we return the entire movie dictionaries (including whatever keys they have) this functionality could be helpful in other waves. So rather than building a new dictionary to return with only the title in the movie dict, we could iterate over the movies in the user watched list and add them to the return collection only if their title appears in the unique list you calculated!